src/compas/artists/artist.py

```diff
@@ -94,7 +117,8 @@ class Artist(object):
 
     def __new__(cls, item, **kwargs):
         if not Artist.__ARTISTS_REGISTERED:
-            register_artists()
+            detected_contexts = cls.register_artists()
```

Comment:
is this not fragile?
should we perhaps not add a function that detects available contexts explicitly?

Reply:
On second thought, it would be more robust to take the detected contexts from `Artist.ITEM_ARTIST`.. that why no changes are needed on the plugin side.